crates/xline/src/utils/trace.rs

```diff
@@ -1,9 +1,22 @@
-use anyhow::Result;
+use anyhow::{Ok, Result};
 use opentelemetry_contrib::trace::exporter::jaeger_json::JaegerJsonExporter;
 use opentelemetry_sdk::runtime::Tokio;
+use tracing::warn;
 use tracing_appender::non_blocking::WorkerGuard;
 use tracing_subscriber::{fmt::format, layer::SubscriberExt, util::SubscriberInitExt, Layer};
-use utils::config::{file_appender, LogConfig, TraceConfig};
+use utils::config::{default_rotation, file_appender, LogConfig, TraceConfig};
+
+/// Return a Box trait from the config
+fn generate_writer(name: &str, log_config: &LogConfig) -> Box<dyn std::io::Write + Send> {
+    if let Some(ref file_path) = *log_config.path() {
+        Box::new(file_appender(*log_config.rotation(), file_path, name))
+    } else {
+        if *log_config.rotation() != default_rotation() {
```

Comment:
default rotation is daily. Do you mean `Never` here?

Reply:
Done